node.gyp

```diff
@@ -959,6 +1019,119 @@
+          }]],
+        }],
+        [ 'v8_enable_inspector==1', {
+          'defines': [
+            'HAVE_INSPECTOR=1',
+          ],
+          'sources': [
+            'src/inspector_agent.cc',
+            'src/inspector_socket.cc',
+            'src/inspector_socket_server.cc',
+            'src/inspector_agent.h',
+            'src/inspector_socket.h',
+            'src/inspector_socket_server.h',
+          ],
+          'dependencies': [
+            'v8_inspector_compress_protocol_json#host',
+          ],
+          'include_dirs': [
+            '<(SHARED_INTERMEDIATE_DIR)/include', # for inspector
+            '<(SHARED_INTERMEDIATE_DIR)',
+          ],
+        }, {
+          'defines': [ 'HAVE_INSPECTOR=0' ]
+        }],
+        [ 'node_use_openssl=="true"', {
+          'defines': [ 'HAVE_OPENSSL=1' ],
+          'sources': [
+            'src/node_crypto.cc',
+            'src/node_crypto_bio.cc',
+            'src/node_crypto_clienthello.cc',
+            'src/node_crypto.h',
+            'src/node_crypto_bio.h',
+            'src/node_crypto_clienthello.h',
+            'src/tls_wrap.cc',
+            'src/tls_wrap.h'
+          ],
+          'conditions': [
+            ['openssl_fips != ""', {
+              'defines': [ 'NODE_FIPS_MODE' ],
+            }],
+            [ 'node_shared_openssl=="false"', {
+              'dependencies': [
+                './deps/openssl/openssl.gyp:openssl',
+
+                # For tests
+                './deps/openssl/openssl.gyp:openssl-cli',
+              ],
+              # Do not let unused OpenSSL symbols to slip away
+              'conditions': [
+                # -force_load or --whole-archive are not applicable for
+                # the static library
+                [ 'node_target_type!="static_library"', {
+                  'xcode_settings': {
+                    'OTHER_LDFLAGS': [
+                      '-Wl,-force_load,<(PRODUCT_DIR)/<(OPENSSL_PRODUCT)',
+                    ],
+                  },
+                  'conditions': [
+                    ['OS in "linux freebsd" and node_shared=="false"', {
+                      'ldflags': [
+                        '-Wl,--whole-archive,'
+                            '<(PRODUCT_DIR)/obj.target/deps/openssl/'
+                            '<(OPENSSL_PRODUCT)',
+                        '-Wl,--no-whole-archive',
+                      ],
+                    }],
+                    # openssl.def is based on zlib.def, zlib symbols
+                    # are always exported.
+                    ['use_openssl_def==1', {
+                      'sources': ['<(SHARED_INTERMEDIATE_DIR)/openssl.def'],
+                    }],
+                    ['OS=="win" and use_openssl_def==0', {
+                      'sources': ['deps/zlib/win32/zlib.def'],
+                    }],
+                  ],
+                }],
+              ],
+            }]]
+        }, {
+          'defines': [ 'HAVE_OPENSSL=0' ]
+        }],
```

Comment:
That's an awful amount of duplication...  There is probably a better way, like moving it into a node.gypi file and including that.